packages/mui-material/src/ButtonBase/ButtonBase.js

```diff
@@ -32,38 +32,46 @@ export const ButtonBaseRoot = styled('button', {
   name: 'MuiButtonBase',
   slot: 'Root',
   overridesResolver: (props, styles) => styles.root,
-})({
-  display: 'inline-flex',
-  alignItems: 'center',
-  justifyContent: 'center',
-  position: 'relative',
-  boxSizing: 'border-box',
-  WebkitTapHighlightColor: 'transparent',
-  backgroundColor: 'transparent', // Reset default value
-  // We disable the focus ring for mouse, touch and keyboard users.
-  outline: 0,
-  border: 0,
-  margin: 0, // Remove the margin in Safari
-  borderRadius: 0,
-  padding: 0, // Remove the padding in Firefox
-  cursor: 'pointer',
-  userSelect: 'none',
-  verticalAlign: 'middle',
-  MozAppearance: 'none', // Reset
-  WebkitAppearance: 'none', // Reset
-  textDecoration: 'none',
-  // So we take precedent over the style of a native <a /> element.
-  color: 'inherit',
-  '&::-moz-focus-inner': {
-    borderStyle: 'none', // Remove Firefox dotted outline.
-  },
-  [`&.${buttonBaseClasses.disabled}`]: {
-    pointerEvents: 'none', // Disable link interactions
-    cursor: 'default',
-  },
-  '@media print': {
-    colorAdjust: 'exact',
-  },
+})(({ ownerState }) => {
+  const { isButton } = ownerState;
+
+  return {
+    display: 'inline-flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    position: 'relative',
+    boxSizing: 'border-box',
+    WebkitTapHighlightColor: 'transparent',
+    backgroundColor: 'transparent', // Reset default value
+    // We disable the focus ring for mouse, touch and keyboard users.
+    outline: 0,
+    border: 0,
+    margin: 0, // Remove the margin in Safari
+    borderRadius: 0,
+    padding: 0, // Remove the padding in Firefox
+    cursor: 'pointer',
+    verticalAlign: 'middle',
+    MozAppearance: 'none', // Reset
+    WebkitAppearance: 'none', // Reset
+    textDecoration: 'none',
+    // So we take precedent over the style of a native <a /> element.
+    color: 'inherit',
+    '&:active': {
+      userSelect: 'none',
+    },
+    '&::-moz-focus-inner': {
+      borderStyle: 'none', // Remove Firefox dotted outline.
+    },
+    [`&.${buttonBaseClasses.disabled}`]: {
+      cursor: 'default',
+      ...(!isButton && {
+        pointerEvents: 'none', // Disable link interactions
```

Comment:
A guess it's why we do this?

```suggestion
        pointerEvents: 'none', // Prevent double click text selection
```

Reply:
It's the other way around. `pointer-events: none` is used here to turn off event callbacks (`on-click`, `on-mouse-down`, etc.) in case the element is not a `button` (as `button` disables them natively when `disabled`).

The side effect is that double-click selection is **enabled**, which is unfortunate. I couldn't find a workaround for this.

So the comment `"Disable link interactions"` is correct. It could be more general `"Disable interactions"` as it does apply for any non `button` element.

Reply:
Clarification: `pointer-events: none` does not enable double-click selection by itself but rather prevents us from handling text selection as we want:

To disable only double-click selection, we can use

```css
.&:active {
    user-select: none
};
```

This way, user selection is only disabled when clicking (`active` state), but not when dragging from outside the element. But with `pointer-events: none`, we don't get the `active` state, so we cannot use this trick.

Using `user-select: none` without the `active` state selector is not a good option as that disables both double-click and selection dragging from outside.

Reply:
Also, `pointer-events: none` does break the native `button` handling of text selection and enables double-click selection for these, so we shouldn't add `pointer-events: none` to `button` elements. It's also not necessary on these, as pointer events are turned off natively.